components/tooltip/script.js

```diff
@@ -90,15 +107,49 @@ class Tooltip extends HTMLElement {
                 break;
             default:
                 position.top = elementSize.top - TOOLTIP_MARGIN - tooltipSize.height;
-                console.log(`The provided option for position ${this.position} is not valid - using top as a fallback. Possible options are top, bottom, left and right.`);
-                this.position = 'top';
+                console.log(`The provided option for position ${orientation} is not valid - using top as a fallback. Possible options are top, bottom, left and right.`);
+                orientation = 'top';
                 break;
         }
 
-        this.style.top = position.top + 'px';
-        this.style.left = position.left + 'px';
-        this.style.visibility = 'visible';
-        this.visible = true;
+        this.position = orientation;
+
+        this.style.top = scrollOffsetY + position.top + 'px';
+        this.style.left = scrollOffsetX + position.left + 'px';
+
+        // check if the tooltip still overflows and re-position it again
+        const overflows  = this.overflows();
+        if(Object.keys(overflows).length && this.fixedSides.length !== 4) {
+            this.setPosition(this.getVisiblePosition(overflows));
+        }
+    }
+
+    overflows() {

```

Comment:
We can change this to a getter to avoid `const overflows = this.overflows()`, but use `this.overflows` directly

Reply:
Changed.